File: in js/stack.js
class stack{
	constructor(){
		this.arr=[];
	}
	push(element)
	{
		this.arr.push(element);
	}
	isEmpty()
	{
		return this.arr.length==0;
	}

	pop()
	{
		if(this.arr.length==0)
			return 'underflow';
		return this.arr.pop();
	}
	peek()
	{

		return this.arr[this.arr.length-1];
	}
};


var s=new stack;
console.log('is stack empty?\n'+s.isEmpty())
s.push(10);
s.push(20);
s.push(30);

console.log('Popping the top element and viewing the top element')
console.log(s.pop())
console.log(s.peek())
console.log('popping again\n')
console.log(s.pop())
console.log('After popping\n'+s.peek())

File: CPP/dsa/basics/stack/stack.cpp
#include <iostream>
#include <stdlib.h>
using namespace std;
#define max 100
typedef struct stack{
	int top;
	int arr[max];
}st;

void createEmptyStack(st *s){
	s->top=-1;
}

int isEmpty(st *s){
	if(s->top == -1)
		return 1;
	else
		return 0;
}

int isFull(st *s){
	if(s->top == max -1)
		return 1;
	else return 0;
}

void push(st *s	,int val){
	if(!isFull(s)){
		s->top++;
		s->arr[s->top]= val;
	}
	else cout << "Stack is Full cant push!!!"<< endl;
}

void pop(st *s){
	if(!isEmpty(s)){
		cout << "The popped out item is= "<< s->arr[s->top] << endl;
		s->top--;
	}
	else cout << "Stack is empty cant pop!!!" << endl;
}
int main(){
	st *s = (st *)malloc(sizeof(st));
	createEmptyStack(s);
	push(s, 10);
	push(s, 11);
	push(s, 12);
	push(s, 13);
	
	pop(s);
	pop(s);

	while (!isEmpty(s)){
		pop(s);
	}


	return 0;
}